torch/csrc/autograd/saved_variable.cpp

```diff
@@ -1,3 +1,4 @@
+#include "Python.h"
```

Comment:
It's due to "_XOPEN_SOURCE" redefined warnings in Python2. The only way to prevent them is to make sure Python.h is included before standard library headers, and the only way to do that is to put it at the top of the compilation unit's include list. It sucks, but I don't know of another way to fix it.

Reply:
I've seen
```
#undef _POSIX_C_SOURCE
#undef _POSIX_THREADS
#undef _XOPEN_SOURCE
```
before `#include <Python.h>` in other projects, e.g. https://github.com/Slicer/VTK/blob/58e382c061ec870de18ef1516db3009c30ae6577/Wrapping/Python/vtkPython.h

If we create a header file that includes `Python.h` after the above `undefs`, we could include that file only where `Python.h` is actually needed.

Reply:
Not sure if it is necessarily applicable here, but I'd like to readvertise a header in PyTorch: `torch/csrc/utils/python_stub.h`

```
#pragma once

struct _object;
typedef _object PyObject;
```

If the only dependence on the Python headers you have is just for declaring pointer to PyObject, you can use this header instead and sidestep the issue.

Reply:
I’ll play with options a bit and submit a PR if I can get rid of the warnings